docs/tables.rst

```diff
@@ -74,7 +74,9 @@ Mutation (modifies table in place)
     Table.move_to_end
     Table.append
     Table.append_column
+    Table.with_column
     Table.relabel
+    Table.with_relabeling
```

Comment:
These belong in the Transformation section because they return a new table instead of mutating the original!